### DISCUSSION TITLE
Does calc support scripts like GNU bc?

### DISCUSSION THREAD

**Discussion by ghost** [lcn2/calc#95] [Q&A] [answered]
GNU bc allows you to use script files. README.md doesn't mention anything about script files. So it seems calc doesn't support this feature. Please correct me if I'm wrong.

**Comment by lcn2** [comment, marked as answer]
Hello @iahung2,

Calc comes a rich set of standard resource files, scripts that perform a wide variety of actions.  See:

```calc
; help resource
```

for information about the many calc standard resource files shipped with calc.

You may find the source to standard resource files in the calc source tree under the `cal/` subdirectory or the share directory if calc is installed (usually under /usr/local/share/calc or /usr/share/calc depending on your OS).

One may default your own functions in calc.  See:

```calc
; help define
```

for how to define your own functions.

One may then put such defined functions into a file and use:

```calc
; read file.cal
```

(or whatever filename: ending the filename with ".cal" is just a convention) to implement your own resource files.

One may write shelll scripts that invoke calc, with the possibility of using your own resource files and/or the standard resource files.  See:

```calc
; help script
```

for examples and see:

```calc
; help cscript
```

for shell script examples.

**Comment by lcn2** [comment]
Examples of scripts:

```calc
; read lucas
; lucas(1,23209)
... test if 1*2^23209-1 is prime .. return 1 if it is prime

; read lucas_chk
; lucas_chk(1000)
... test various primes of the form h*2^n-1 for n < 1000

; read pi
; piforever()
... print more and more digits of pi .. until you run out of memory or patience :)
```

etc.

All of the standard calc resource scripts are available for viewingtree under the cal/ source subdirectory  (usually under /usr/local/share/calc or /usr/share/calc depending on your OS).

**Reply by lcn2** [reply]
A word of advice: while calc is "C-like" it is NOT C.  Read the unexpected help file:

```calc
; help unexpected
```

For more help on help see:

```calc
; help help
```

And for those who prefer the "man" to "help", "man is an alias for "help":

```calc
; man man
; man overview
```

We hope this helps!
